src/validation/3.0/invalid-property-value.rule.ts

```diff
@@ -138,6 +145,101 @@ export class Oas30InvalidPropertyValueValidationRule extends Oas30PathValidation
         return method === "put" || method === "post" || method === "options" || method === "patch";
     }
 
+    /**
+     * Returns resolved parameter definition or the address of the reference if parameter can not be resolved
+     * (e.g. when the pointer is to a non-existent value within the components).
+     * @param node - Parameter definition within a path item or operation
+     * @return {Oas30Parameter | string} - Returns (recursively) resolved parameter or a string with invalid ref.
+     */
+    private resolveParameter(node: Oas30Parameter): Oas30Parameter | string {
```

Comment:
We can certainly move the recursion to ReferenceUtil - will have to take a closer look at the class. On a related note, the fact that OpenApi supports indirect references, there is always a possibility of circular references. For example:

"components": {
  "parameters": {
    "ref1": { "$ref": "#/components/parameters/ref2"},
    "ref2": { "$ref": "#/components/parameters/ref1"}
  }
}

I had actually written code in my validator to check for such circular references (by maintaining a list of all the references it has encountered and bailing out if the same it was asked to visit the same reference again) but removed it because I think it doesn't even get to the point of validation - from what I can tell the js2model.reader goes into an infinite loop when creating model objects.

Reply:
Created a new generic method in ReferenceUtils.ts to resolve references recursively. It returns the ref pointer of the last failed resolution instead of a null. I have explained the reasoning behind it in answer to your comment 1) above.